onnxruntime/core/mlas/lib/amx_common.h

```diff
@@ -42,9 +44,33 @@ tile_loadconfig(const void* __config)
 }
 #else
 
-#define tile_int8_dp_internal(name,dst,src1,src2)					\
-  __asm__ volatile							\
-  ("{"#name"\t%%tmm"#src2", %%tmm"#src1", %%tmm"#dst"|"#name"\t%%tmm"#dst", %%tmm"#src1", %%tmm"#src2"}" ::)
+extern "C" { 
+	void MlasTdpAmx(int dst, int src1, int src2); 
+	void MlasTdpAmxT4T2T0(int dst, int src1, int src2); 
+	void MlasTdpAmxT5T3T0(int dst, int src1, int src2); 
+	void MlasTdpAmxT6T2T1(int dst, int src1, int src2); 
+	void MlasTdpAmxT7T3T1(int dst, int src1, int src2); 
+}
+//MlasTdpAmx(dst,src1,src2)
+
+#define tile_dpbusd_t4t2t0(dst,src1,src2)			\
```

Comment:
So you need 8x8x8 = 512 different functions, and since you've already specified all the parameters in the function name, why do you need those parameters? This is just not acceptable.

Reply:
I have updated the code for the instruction macros to be directly called from C++ and doesn't need the enumeration of all the cases from all  the tile id.